### ISSUE TITLE
Security descriptor issue when running passthrough with network backend

### ISSUE THREAD

**Issue by earonesty** [billziss-gh/winfsp#215]
# system

We tested this on win7 and win10

## To reproduice

1. build the passthrough file system (i used passthrough-cpp with latest master, and latest winfsp release installed)

2. run a passthrough system.   it doesn't matter if you use drive or folder mounts 
`./build/Debug/passthrough-cpp-x64.exe -p 'z:\' -m 'c:\ptnet'`

3. copy a folder full of data to the passthrough `c:\ptnet\stuff\...`

4. using windows explorer, copy and paste the folder... so that it tries to overwrite itself.   normal behavior is for it to create  `c:\ptnet\stuff - Copy`.   Instead you get an odd error (File name already exists).    The cause of this error is a concern however.   When looking at the procmon logs, we see an "invalid security descriptor" error.    The problem occurs when trying to pass the local descriptor to the network mounted file (and vice versa).   

5. The specific error is `STATUS_INVALID_OWNER`, and it occurs during the "Read Directory" operation that explorer runs right after creating the empty copy folder.

**Comment by billziss-gh** [comment]
I should note here that the passthrough file systems in general have multiple problems with access control because they run as normal user processes and they are subjected to access checks by the underlying file system. One way to mitigate (but not eliminate) some of these problems is to run them under an account that has the BACKUP and RESTORE privileges, which the passthrough file system attempts to [enable on start](https://github.com/billziss-gh/winfsp/blob/v1.4/tst/passthrough/passthrough.c#L739-L769).

Furthermore the C++ layer and the passthrough-cpp file system is not shipped in the installer and has not been fully tested (because I never fully finished the C++ layer).

I am wondering: does this problem happen with normal passthrough?

**Comment by earonesty** [comment]
Yes, this problem also occurs with the C, layer, with the same STATUS_INVALID_OWNER issue
